websites/F/Fandom/presence.ts

```diff
@@ -1,401 +1,407 @@
-if (
-	document.location.pathname.includes("/wiki/")
-		? document.querySelector(".skin-oasis") ||
-		  ((document.querySelector(".skin-fandomdesktop") ||
-				document.querySelector(".skin-fandommobile")) &&
-				!document.querySelector(".is-gamepedia"))
-		: true
-) {
-	((): void => {
-		const presence = new Presence({
-				clientId: "644400074008297512",
-			}),
-			browsingTimestamp = Math.floor(Date.now() / 1000);
-		let currentURL = new URL(document.location.href),
-			currentPath = currentURL.pathname.replace(/^\/|\/$/g, "").split("/"),
-			presenceData: PresenceData = {
-				details: "Viewing an unsupported page",
-				largeImageKey: "lg",
-				startTimestamp: browsingTimestamp,
-			};
-		const updateCallback = {
-				_function: null as () => void,
-				get function(): () => void {
-					return this._function;
-				},
-				set function(parameter) {
-					this._function = parameter;
-				},
-				get present(): boolean {
-					return this._function !== null;
-				},
-			},
-			/**
-			 * Initialize/reset presenceData.
-			 */
-			resetData = (
-				defaultData: PresenceData = {
-					details: "Viewing an unsupported page",
-					largeImageKey: "lg",
-					startTimestamp: browsingTimestamp,
-				}
-			): void => {
-				currentURL = new URL(document.location.href);
-				currentPath = currentURL.pathname.replace(/^\/|\/$/g, "").split("/");
-				presenceData = { ...defaultData };
-			},
-			/**
-			 * Search for URL parameters.
-			 * @param urlParam The parameter that you want to know about the value.
-			 */
-			getURLParam = (urlParam: string): string => {
-				return currentURL.searchParams.get(urlParam);
-			};
+const presence = new Presence({
+		clientId: "644400074008297512",
+	}),
+	browsingStamp = Math.floor(Date.now() / 1000);
+let currentURL = new URL(document.location.href),
+	currentPath = currentURL.pathname.replace(/^\/|\/$/g, "").split("/"),
+	presenceData: PresenceData = {
+		details: "Viewing an unsupported page",
+		largeImageKey: "lg",
+		startTimestamp: browsingStamp,
+	};
+const updateCallback = {
+		_function: () => void {} as () => void,
+		get function(): () => void {
+			return this._function;
+		},
+		set function(parameter) {
+			this._function = parameter;
+		},
+	},
+	/**
+	 * Initialize/reset presenceData.
+	 */
+	resetData = (
+		defaultData: PresenceData = {
+			details: "Viewing an unsupported page",
+			largeImageKey: "lg",
+			startTimestamp: browsingStamp,
+		}
+	): void => {
+		currentURL = new URL(document.location.href);
+		currentPath = currentURL.pathname.replace(/^\/|\/$/g, "").split("/");
+		presenceData = { ...defaultData };
+	},
```

Comment:
Alright, this goes for all the PRs. This seems like an overly sophisticated method of completing a fairly simple task which is fine normally, but as the Presence repo is open-source this needs to be widely understandable and I haven't got a clue on why you have added this level of confusion to a very simple function. 
Please refactor this and make it understandable for everyone.

Reply:
I know this is subjective, but I think confusion is an overstatement 

The resetData function is to reset the presence data to the state after it is prepared. 

After the presence data has been prepared (on the `prepare()` function), the usual way I did is to send the data directly, skipping the preparation part (and an optimal way than conventional ways). However, there are some cases when after the preparation, the page doesn't refresh but changes in some way, you need to update the presence again. For Fandom, this is when a user wants to edit the page with VisualEditor. For that, the `updateCallback()` is made to store the functions on what to do after the preparation. This also true if you want to hide the buttons and the timestamps using the settings.

The way it works is that the function modifies the **prepared** presence data, instead of start over with the blank page, also skipping the heavier parts. With that, on the `presence.on("UpdateData", () => {})` callback, the presence data would be reset to the **prepared** state before the `updateCallback()` can modify it, along with the buttons and timestamps can be hidden based on the settings.

Now it's been a while since I created this code base (and also debated), and with the use case of the MediaWiki sites, I have kind of simplified it. I have simplified this part of code enough, and I hope it is clear enough now.

I'm unsure if this is also your point, but I putting this here: However, it is my best interest to not compromise the quality of the code/logic in favor of readability/understandability. I do recognize a better way to do this: documentation in form of code comments. I would do this within your comment about this.

[Related write-up of my code base, but it's from 2020 so a lot has changed since then.](https://hans5958.github.io/blog/2020/06/29/explanation-for-my-presence-ts-base-on-premid)

Reply:
The confusion is very valid, other's have also questioned it too. I kind of see why you are doing this, but from a maintainability perspective this presence is poor because it isn't really inline with other presences which makes it incredibly hard to maintain